shell/platform/darwin/ios/BUILD.gn

```diff
@@ -263,7 +263,6 @@ source_set("ios_test_flutter_mrc") {
 }
 
 shared_library("ios_test_flutter") {
-  testonly = true
```

Comment:
Why are we getting rid of this? It seems like a useful guard against non-test targets including test TUs.

Reply:
When this flag is on, because `flutter_framework` is not testonly, it cannot depend on `ios_test_flutter`

Reply:
We should not make flutter_framework depend on ios_test_flutter, we should update some higher level `BUILD.gn` that checks if unit tests are enabled and if so include ios_test_flutter (if ios etc.)

Reply:
e.g. in `//BUILD.gn` something like

```gn
if (enable_unittests && is_ios) {
  public_deps += [ "//flutter/shell/platform/darwin/ios:ios_test_flutter" ]
}
```

Reply:
Can we add `testonly= true` back now?